### PR TITLE
feat(insured-bridge): Update Arbitrum messenger contracts after e2e tests

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

Arbitrum L1 --> L2 messaging works differently in practice than originally assumed. According to [these docs](https://developer.offchainlabs.com/docs/l1_l2_messages#retryable-transaction-lifecycle), the L2 sender, which in our case is the [_aliased address_](https://developer.offchainlabs.com/docs/l1_l2_messages#address-aliasing) of the `ArbitrumMessenger`, needs to pay for the "submission fee". Additionally, the L2 gas limit + price needs to be specified correctly for the L2 transaction to immediately succeed. 

In the section below, I detail the entire L1 --> L2 successful flow where I updated the L2 minimum bridging delay from 60 --> 55.

**Test Summary**

Deploy and setup contracts on L1 (Ethereum Rinkeby):
- [Deploy Finder](https://rinkeby.etherscan.io/address/0xbb6206fb01fAad31e8aaFc3AD303cEA89D8c8157#code)
- [Deploy IDWhitelist](https://rinkeby.etherscan.io/address/0xCF744280D6ee7ca04e6D4167546a8463Ad899105#code)
- [Deploy CollateralWhitelist](https://rinkeby.etherscan.io/address/0xbBD1B575BEE5A77d393973Fe79B1d6e1762BB1d0#code)
- [Set IDWhitelist in Finder](https://rinkeby.etherscan.io/tx/0x2b9af319822543729c98b125af357414ba4c6e3aab695863679cba376a811272)
- [Set CollateralWhitelist in Finder](https://rinkeby.etherscan.io/tx/0x539f3a5777f9c524de8d5455970d814b059e3e857a519a86288638211bc2bb68)
- [Whitelist “IS_CROSS_CHAIN_RELAY_VALID” in IDWhitelist](https://rinkeby.etherscan.io/tx/0xe8bbcebd5f1eca6174ec95d4ff3cf94f654d7ba162cda43f3408b6d1df3237ef)
- [Deploy BridgeAdmin using identifier that we just whitelisted](https://rinkeby.etherscan.io/address/0x040A1E425D5C5663018414C9100C86ab16fa8Fce#code)
- [Deploy Arbitrum_Messenger using predeployed inbox @ 0x578BAde599406A8fE3d24Fd7f7211c0911F5B29e](https://rinkeby.etherscan.io/address/0x7EACB85A07ACbF44Ea6600A93BC0C11E130E49fB#code)
- [Set Arbitrum_Messenger owner to BridgeAdmin](https://rinkeby.etherscan.io/tx/0x167a1d90fee47ecf941be81d29cb42ce77113f9955d3e451cfe9f22ab936b471)

Deploy and setup contracts on L2 (Arbitrum Rinkeby):
- [Deploy AVM_BridgeDepositBox using params from https://developer.offchainlabs.com/docs/public_testnet and setting above ArbitrumMessenger as crossDomainAdmin](https://testnet.arbiscan.io/address/0xd2ecb3afe598b746F8123CaE365a598DA831A449#code)

Additional setup onL1:
- [Set deposit contract for chainID on BridgeAdmin](https://rinkeby.etherscan.io/tx/0xa7f17d6f64b1908f77d0bf82c7b2b403c5c7ab8d64b87da52c41371362bd84fe)

Set minimum bridging delay on L2 from L1:
- [`BridgeAdmin.setMinimumBridgingDelay` should send cross chain message to `AVM_DepositBox`](https://rinkeby.etherscan.io/tx/0xdfac063dd7d97baf09a2ce19539f57c74d60170f005831039f4c929f3607e67f)
- Setting the L2 retryable parameters is tricky, but I followed the steps in this [tutorial](https://github.com/OffchainLabs/arbitrum-tutorials/blob/master/packages/greeter/scripts/exec.js)
- First, compute `maxSubmissionCost`, which is a function of the `setMinimumBridgingDelay` calldata. I safely overestimated this as 1e18 wei.
- Next, guesstimate the `gasPriceBid` and `l2GasLimit` based on the example script which was 0.1 gwei and 1,000,000 gas respectively. It's important to set these correctly or the ticket will not immediately execute.
- Finally, determine how much ETH needs to be sent to L2 in order to pay for the total callvalue, which will also be set as the l1CallValue. This can be computed deterministically as `maxSubmissionCost + gasPrice * l2GasLimit`. 
- Submit the L1 transaction and set the `msg.value` equal to `l1CallValue`.
- After some time, the aliased `msg.sender` will receive the `l1CallValue` and submit the transaction. The aliased msg.sender is the aliased ArbitrumMessenger address + the offset: 0x111100…001111, which can be seen [here](https://testnet.arbiscan.io/address/0x8fbdb85a07acbf44ea6600a93bc0c11e130e5b0c)
- The three L2 transactions are: RetryableTicket is created, RetryableTicket is auto redeemed, transaction to BridgeDepositBox is executed.
- [This address will submit the L2 `setMinimumBridgingDelay` txn](https://testnet.arbiscan.io/tx/0x090e1133668e8cc8584395c3a506a7be4346b183754dcf0b84dbb7e63632b006)
- Note: all of the above function params can be computed using the `arb-ts` node package

Useful tools for debugging:
- [rough dashboard](https://retryable-dashboard.netlify.app/)

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts/insured-bridge/ovm/Optimism_CrossDomainEnabled.sol [hunk 1]
```diff
@@ -76,7 +76,7 @@ contract Optimism_CrossDomainEnabled {
      */
     function sendCrossDomainMessage(
         address _crossDomainTarget,
-        uint32 _gasLimit,
+        uint256 _gasLimit,
```
Comment (nicholaspai) [inline packages/core/contracts/insured-bridge/ovm/Optimism_CrossDomainEnabled.sol:79]: Just making these uint lengths consistent with the Arbitrum messenger's. I'm not worried about gas costs for BridgeAdmin transactions which are not user-facing and will not be frequently called.

**Review by chrismaree — APPROVED** [review]
This took some serious detective work! great job @nicholaspai !

**Review by mrice32 — APPROVED** [review]
LGTM! Very nice work. This seems pretty complicated. Anything we should watch out for or ways that things maybe could get stuck, broken, or attacked?

File: packages/core/contracts/insured-bridge/avm/Arbitrum_Messenger.sol [hunk 2]
```diff
@@ -19,22 +21,29 @@ contract Arbitrum_Messenger is Ownable, Arbitrum_CrossDomainEnabled, MessengerIn
...
         bytes memory message
-    ) external override onlyOwner {
+    ) external payable override onlyOwner {
         // Since we know the L2 target's address in advance, we don't need to alias an L1 address.
         sendTxToL2NoAliassing(
             target,
             target, // send any excess ether to the L2 deposit box.
-            0, // TODO: Determine the max submission cost. From the docs: "current base submission fee is queryable via ArbRetryableTx.getSubmissionPrice"
+            l1CallValue,
+            maxSubmissionCost, // TODO: Determine the max submission cost. From the docs: "current base submission fee is queryable via ArbRetryableTx.getSubmissionPrice"
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply packages/core/contracts/insured-bridge/avm/Arbitrum_Messenger.sol:46]: Yeah I'm going to send the refund to the original `msg.sender` of the BridgeAdmin call

File: packages/core/contracts/insured-bridge/test/OVM_L1CrossDomainMessengerMock.sol [hunk 3]
```diff
@@ -0,0 +1,17 @@
+pragma solidity ^0.8.0;
+import "../ovm/iOptimism_CrossDomainMessenger.sol";
+
+contract OVM_L1CrossDomainMessengerMock is iOptimism_CrossDomainMessenger {
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply packages/core/contracts/insured-bridge/test/OVM_L1CrossDomainMessengerMock.sol:4]: I was getting some weird reverts on `smockit` with this contract so I pulled it into the repo

**Pull request merged**